sbi/src/api/tsm.rs

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) 2022 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::TeeFunction::*; // TODO
+use crate::{ecall_send, Error, Result, SbiMessage};
+use crate::{TsmInfo, TsmPageType, TvmCpuRegister, TvmCreateParams}; //TODO
+
+/// Initiates a TSM fence on this CPU.
+pub fn initiate_fence() -> Result<()> {
+    let msg = SbiMessage::Tee(TsmInitiateFence);
+    // Safety: TsmInitiateFence doesn't read or write any memory we have access to.
+    unsafe { ecall_send(&msg).map(|_| ()) }
```

Comment:
The `ecall_send(&msg).map(..)` pattern gets repeated a lot -- maybe something like this is cleaner?

```
unsafe { ecall_send(&msg) }?;
Ok(())
```

Reply:
good idea, done.